src/CourseService.php

```diff
@@ -386,6 +386,89 @@ public function UpdateAssetsFromUploadedFile($courseId, $uploadLocation)
     	
     	return $response;
     }
+
+    /// <summary>
+    /// Import a SCORM .pif (zip file) from the local filesystem using async client.
+    /// </summary>
+    /// <param name="courseId">Unique Identifier for this course.</param>
+    /// <param name="absoluteFilePathToZip">Full path to the .zip file</param>
+    /// <returns>List of Import Results</returns>
+    public function ImportCourseAsync($courseId, $absoluteFilePathToZip)
+    {
+        $uploadService = new UploadService($this->_configuration);
+        $location = $uploadService->UploadFile($absoluteFilePathToZip);
+
+        $importException = null;
+        $response = null;
+        try {
+            $token = $this->ImportCourseAsyncToken($courseId, $location);
+            $done = false;
+            while (!$done){
+                $xmlStatus = $this->getAsyncImportStatus($token);
+                $statusResult = (string) $xmlStatus->status;
+                switch ($statusResult) {
+                    case 'running':
```

Comment:
make these as const.

Reply:
so is response an array or an object?

Reply:
You mean the response from the web service? Then it is an xml, not an object. I have to convert it to xml element to fetch the status